syft/workers/base.py

```diff
@@ -94,8 +84,26 @@ def __init__(
             MSGTYPE.GET_SHAPE: self.get_tensor_shape,
             MSGTYPE.SEARCH: self.search,
         }
+
         self.load_data(data)
 
+        # Declare workers as appropriate
+        self._known_workers = {}
+        if hook.local_worker is not None:
+            known_workers = self.hook.local_worker._known_workers
+            if self.id in known_workers:
+                if isinstance(known_workers[self.id], type(self)):
+                    self.__dict__.update(known_workers[self.id].__dict__)
```

Comment:
What exactly is this line doing? Could you add a comment

Reply:
If you think more details would be helpful I can try to add a better explanation.